tests/tools/saunafs.sh

```diff
@@ -449,48 +449,75 @@ prepare_metalogger_() {
 	saunafs_info_[metalogger_cfg]="$etcdir/sfsmetalogger.cfg"
 }
 
-create_sfshdd_cfg_() {
-	local n=$disks_per_chunkserver
-	local zoned_prefix=""
+# Creates the needed emulated zoned devices for a given chunkserver number.
+#
+# The function also creates the needed directories for the metadata and data
+# paths of the chunkserver. The metadata path is created in the ramdisk and the
+# data path is created in the zoned device. The function also formats and mounts
+# the zoned device using the zonefs filesystem.
+#
+# Parameters:
+# - chunkserver_index (optional): The chunkserver number.
+#   Defaults to "chunkserver_id".
+#
+# Usage:
+# create_emulated_devices_for_cs_number_ [chunkserver_index]
+#
+create_emulated_zoned_devices_for_cs_number_() {
+	if [[ ! $use_zoned_disks || ! $use_ramdisk ]]; then
+        return
+    fi
+
+	local chunkserver_index=${1:-chunkserver_id}
+	local n=${disks_per_chunkserver}
+	local zoned_prefix="zonefs:"
+	local firstDisk=$(echo "${chunkserver_index} * ${n}" | bc)
+	local lastDisk=$(echo "${firstDisk} + ${n} - 1" | bc)
+
+	for disk_number in $(seq -w ${firstDisk} ${lastDisk}); do
+		if mount | grep "sauna_nullb${disk_number}" &>/dev/null; then
+			sudo umount -l "/mnt/zoned/sauna_nullb${disk_number}" &>/dev/null
+		fi
 
-	if [[ $use_zoned_disks && $use_ramdisk ]]; then
-		zoned_prefix="zonefs:"
+		create_zoned_nullb 4096 $zone_size_mb $number_of_conv_zones \
+			$number_of_seq_zones ${disk_number}
 
-		local metadataPath
-		local dataPath
-		local devPath
-		local firstDisk=$(echo "${chunkserver_id} * ${n}" | bc)
-		local lastDisk=$(echo "${firstDisk} + ${n} - 1" | bc)
+		if [ -b /dev/sauna_nullb${disk_number} ]; then
+			local devPath="/dev/sauna_nullb${disk_number}"
+		elif [ -b /dev/nullb${disk_number} ]; then
+			local devPath="/dev/nullb${disk_number}"
+		fi
 
-		for disk_number in $(seq -w ${firstDisk} ${lastDisk}); do
-			if mount | grep "sauna_nullb${disk_number}" &>/dev/null; then
-				sudo umount -l "/mnt/zoned/sauna_nullb${disk_number}" &>/dev/null
-			fi
+		sudo mkzonefs -f -o perm=666 "${devPath}" &>/dev/null
 
-			create_zoned_nullb 4096 $zone_size_mb $number_of_conv_zones \
-				$number_of_seq_zones ${disk_number}
+		local metadataPath="/mnt/ramdisk/metadata/sauna_nullb${disk_number}"
+		local dataPath="/mnt/zoned/sauna_nullb${disk_number}"
 
-			if [ -b /dev/sauna_nullb${disk_number} ]; then
-				devPath="/dev/sauna_nullb${disk_number}"
-			elif [ -b /dev/nullb${disk_number} ]; then
-				devPath="/dev/nullb${disk_number}"
-			fi
+		if [ ! -d "${metadataPath}" ]; then
+			mkdir -pm 777 "${metadataPath}"
+		fi
 
-			sudo mkzonefs -f -o perm=666 "${devPath}" &>/dev/null
+		# Mount point
+		if [ ! -d "${dataPath}" ]; then
+			sudo mkdir -pm 777 "${dataPath}"
+		fi
 
-			metadataPath="/mnt/ramdisk/metadata/sauna_nullb${disk_number}"
-			dataPath="/mnt/zoned/sauna_nullb${disk_number}"
+		sudo mount -t zonefs "${devPath}" "${dataPath}"
+	done
+}
```

Comment:
I'd try to minimize root usage as much as possible. Why not put the dataPath under the /tmp/ directory or directories that saunafs-tests already uses?

Reply:
I agree, but probably that's for another change. In this change I am just separating the functions and using similar places as the setup_machine uses for ramdisk, loop devices, etc.